redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -2,44 +2,46 @@ package zio.redis
 
 import java.time.Instant
 import java.util.concurrent.TimeUnit
-
 import scala.util.matching.Regex
-
 import zio.Chunk
 import zio.duration.Duration
+import zio.schema.codec.Codec
+import zio.schema.Schema
 
 sealed trait Input[-A] {
-  private[redis] def encode(data: A): Chunk[RespValue.BulkString]
+  private[redis] def encode(data: A)(implicit codec: Codec): Chunk[RespValue.BulkString]
```

Comment:
I wouldn't pass it implicitly, please change to the encoding suggested in our chat:

```scala
def encode(data: A, codec: Codec): Chunk[...]
```

Reply:
Ok)
I want to mention that if it is not implicit parameter, then we have to put codec parameter in every encode method inside that one
`final case class Tuple5[-A, -B, -C, -D, -E](_1: Input[A], _2: Input[B], _3: Input[C], _4: Input[D], _5: Input[E])
      extends Input[(A, B, C, D, E)] {
    def encode(data: (A, B, C, D, E))(implicit codec: Codec): Chunk[RespValue.BulkString] =
      _1.encode(data._1) ++ _2.encode(data._2) ++ _3.encode(data._3) ++ _4.encode(data._4) ++ _5.encode(data._5)
  }`

Reply:
I see, let's keep it then. We'll deal with compat later.